### ISSUE TITLE
code improvement suggestions

### ISSUE THREAD

**Issue by abraxxa** [HGrigorov/checkpoint#1]
Hi Hristo,
say you script on Checkmates and want to suggest some improvements as I'm a 20 year+ Perl developer and Checkpoint admin:

- Perl is spelled this way ;-)
- `use lib qw(..);` introduces a security vulnerability which is why the current directory isn't included in the library search path since version 5.26, see https://perldoc.perl.org/perl5260delta#Removal-of-the-current-directory-(%22.%22)-from-@INC
- assigning `undef` to a variable is unneeded typing as this is the default value of each variable in Perl.
- you shouldn't use `our` to define a variable instead of `my` if you're not exactly sure what it's for. In your case `my` is the correct way because it creates a lexical var.
- you can use `say "sometext"` instead of `print "sometext\n"` since Perl version 5.10 (released end of 2007, even Checkpoint should ship a version that's newer by now) if you `use feature 'say';` or `use v5.10;`.
- you might note that the JSON::PP (Pure-Perl) module is shipped with Perl since the stable version 5.14

**Comment by HGrigorov** [comment]
Hello Alex,

Many thanx for your feedback! I am very impressed someone actually looked
at my code and took the time to comment on it. Much appreciated.

The truth is that some of the code I borrowed from an old scripts I wrote
back in the days, some just quickly copied from examples on Internet and
some are just me being lazy or lacking of knowledge about modern Perl
language :)

Anyway, no more excuses! I will try to fix this and that tomorrow but if
you are willing to contribute directly I can give you access and you do it
yourself. I have few ideas how to extend the scripts but just need a little
bit more time...

All the best,
Hristo.
